### ISSUE TITLE
Support bucket expiration via plan or instance parameters

### ISSUE THREAD

**Issue by spiegela** [thecodeteam/ecs-cf-service-broker#48]
AWS S3 bucket lifecycle properties support object/bucket expiration. The broker should be able to support this as a instance parameter provided during the service instance creation/upgrade.  Support for this might look something like:

``` shellsession
$ cf create-service ecs-bucket 5gb temp-bucket -c "{ expiration: {days: 30} }"
```

This could also be supported in the catalog configuration for a service.  We could add something along the lines of:

``` yaml
catalog:
  services:
    - id: <Service ID UUID>
      name: ecs-bucket
      description: Elastic Cloud S3 Object Storage Bucket
      service-settings:
        service-type: bucket
        expiration:
          days: 30
```

In the case of YAML catalog configuration, _I think_ the expiration should be come _locked_ by the service, meaning that it would override a setting attempted during the service-instance creation.

Need to get some more clarification on this aspect...

**Comment by spiegela** [comment]
@discoverkr77 Could you comment on the proposal above?  Does this meet your requirements?

I'm thinking that it would be nice to have the _Broker Administrator_ be able to set the setting globally for a service (as in the 2nd example above), and have that take precedence over anything a _CF user_ specifies.  However, if the expiration is not set in the catalog configuration, then the _CF user_ can set it using the service-instance parameters as shown in the first example.

**Comment by spiegela** [comment]
This feature is somewhat tricky since it requires the S3 client to perform these changes.  It'll be done after #45 & #44, and maybe after some other issues in the backlog.
